src/api/remote.rs

```diff
@@ -151,6 +178,78 @@ impl Integration {
     pub fn endpoint(&self) -> &Remote {
         self.protocol().endpoint()
     }
+
+    /// Best effort approach to set primary branch by searching for main/master branch
+    pub async fn set_watched_branches(&self) -> Result<Integration, Report<ValidationError>> {
+        match self {
+            Integration {
+                poll_interval,
+                team,
+                title,
+                protocol,
+                import_branches: true,
+                import_tags,
+                watched_branches,
+            } => {
+                if !watched_branches.is_empty() {
+                    return Ok(self.clone());
+                }
+
+                let references = self.references().await.unwrap_or_default();
+                let primary_branch = references
+                    .iter()
+                    .find(|r| r.name() == MAIN_BRANCH || r.name() == MASTER_BRANCH)
+                    .cloned();
+                match primary_branch {
+                    None => {
+                        report!(ValidationError::PrimaryBranch)
+                        .wrap_err()
+                        .help("watched_branches was empty and failed to inject main/master branch into watched_branches")
+                        .describe_lazy(||"provide valid watched_branches")?
+                    }
+                    Some(branch) => {
+                        let primary_branch_name = branch.name();
+                        warn!("Watched_branches was set to empty, added branch '{primary_branch_name}' as a best effort approach");
+                        let watched_branch = WatchedBranch::new(branch.name().to_string());
+                        let watched_branches = vec![watched_branch];
+
+                        Integration {
+                            poll_interval: *poll_interval,
+                            team: team.clone(),
+                            title: title.clone(),
+                            protocol: protocol.clone(),
+                            import_branches: true,
+                            import_tags: *import_tags,
+                            watched_branches,
+                        }.wrap_ok()
+                    }
+                }
+            }
+            _ => Ok(self.clone()),
+        }
+    }
+
+    /// Checks if the reference branch should be scanned by comparing it to our watched branches
+    pub fn validate_reference_scan(&self, reference: &str) -> bool {
+        let branches = self.watched_branches();
+        for branch in branches {
+            match Pattern::new(branch.name().as_str()) {
+                Ok(p) => {
+                    if p.matches(reference) {
+                        return true;
+                    }
+                }
+                // In the case of error continue on and have the function return false if there are no matches
+                Err(_e) => continue,
+            }
+        }
+        false
+    }
```

Comment:
[nit, naming] Prefer a function name that more directly reads its intended use, e.g. `should_scan_reference`. Consider its intended callsite: `if should_scan_reference()` vs `if validate_reference_scan`.